test/gtest/ucs/test_async.cc

```diff
@@ -494,6 +494,29 @@ UCS_TEST_P(test_async, ctx_timer_block) {
     EXPECT_GE(lt.count(), 1); /* Timer could expire again after unblock */
 }
 
+UCS_TEST_P(test_async, modify_event) {
+    local_event le(GetParam());
+
+    le.push_event();
+    suspend_and_poll(&le, COUNT);
+    EXPECT_GE(le.count(), 1);
+
+    ucs_async_modify_handler(le.event_id(), 0);
+    le.push_event();
+    suspend_and_poll(&le, COUNT);
+    EXPECT_GE(le.count(), 0);
```

Comment:
I guess you want to check that le.count did not change. Seems that you need to compare for equality with previous counter value